docker/Makefile

```diff
@@ -169,154 +132,118 @@ code vscode: | docker-vscode check-docker
 
-TARGET_distclean += distclean-docker-vscode
-HELP_distclean-docker-vscode = removes vscode docker image
-.PHONY: distclean-docker-vscode
-distclean-docker-vscode: clean-docker-vscode
+.PHONY: distclean-vscode
+DISTCLEAN += distclean-vscode
+HELP_distclean-vscode = removes vscode docker image
+distclean-vscode: clean-vscode
 	-docker image remove "lifesensor/vscode:latest"
+	docker image prune
 
-.PHONY: clean-check
-clean-check:
-	@true
-
-TARGET_distclean += distclean-check
-HELP_distclean-check = removes check result if docker is installed
-.PHONY: distclean-check
-distclean-check: distclean-check-docker
-
-.PHONY: distclean-check-docker
-distclean-check-docker: distclean-check-docker-installed distclean-check-docker-group
-
-.PHONY: distclean-check-docker-installed
-distclean-check-docker-installed:
-	rm -f .check-docker-installed
-
-.PHONY: distclean-check-docker-group
-distclean-check-docker-group:
-	rm -f .check-docker-group
+### docker check installation targets ###
 
 .PHONY: check-docker
+CHECK += check-docker
 check-docker: check-docker-installed check-docker-group
 
 .PHONY: check-docker-installed
-check-docker-installed: | ./.check-docker-installed
-
-./.check-docker-installed:
-	@if which docker > /dev/null 2>&1; \
-	then \
-		touch $@; \
-	else \
-		( \
-		echo "########################################"; \
-		echo "# DOCKER DOES NOT SEEM TO BE INSTALLED #"; \
-		echo "########################################"; \
-		echo "the docker binary could not be found in your PATH!"; \
-		echo "This usally means docker is simply not installed."; \
-		echo -n "I detected your platform as "; \
-		if cat /etc/*-release | grep -qi debian; \
-		then \
-			echo DEBIAN, so please run; \
-			echo; \
-			make -sn install-docker-debian; \
-			echo; \
-			echo OR simply; \
-			echo; \
-			echo make install-docker-debian; \
-			echo; \
-		elif cat /etc/*-release | grep -qi ubuntu ; \
-		then \
-			echo UBUNTU, so please run; \
-			echo; \
-			make -sn install-docker-ubuntu; \
-			echo; \
-			echo OR simply; \
-			echo; \
-			echo make install-docker-ubuntu; \
-			echo; \
-		elif cat /etc/*-release | grep -qi "arch linux" ; \
-		then \
-			echo ARCHLINUX, so please run; \
-			echo; \
-			make -sn install-docker-archlinux; \
-			echo; \
-			echo OR simply; \
-			echo; \
-			echo make install-docker-archlinux; \
-			echo; \
-		elif cat /etc/*-release | grep -qi centos ; \
-		then \
-			echo CENTOS, so please run; \
-			echo; \
-			make -sn install-docker-centos; \
-			echo; \
-			echo OR simply; \
-			echo; \
-			echo make install-docker-centos; \
-			echo; \
-		elif cat /etc/*-release | grep -qi fedora ; \
-		then \
-			echo FEDORA, so please run; \
-			echo; \
-			make -sn install-docker-fedora; \
-			echo; \
-			echo OR simply; \
-			echo; \
-			echo make install-docker-fedora; \
-			echo; \
-		else \
-			echo UNKNOWN; \
-			echo please run your package manager to install docker; \
-		fi; \
-		) 1>&2; \
-		exit 1; \
-	fi
+check-docker-installed:
```

Comment:
I am still not convinced we should check for requirements on the software side, but wont block this PR based on it.

Reply:
this is sth we have to discuss outside of this refactor PR